### PR TITLE
Add codespell CI test and fix codespell error

### PR DESCRIPTION
**Pull request by jenting**
#### Proposed Changes ####

Fix the code spell error reported from https://github.com/codespell-project/codespell.

#### Types of Changes ####

- Add a code spell test in CI `./scripts/codespell.sh`
- Correct code spelling

#### Verification ####

None

#### Linked Issues ####

None

#### Further Comments ####

Not sure what's the `ba` means in https://github.com/k3s-io/k3s/blob/4317a91/scripts/provision/vagrant#L54
So I add ignore `ba` in https://github.com/k3s-io/k3s/blob/d64b687223aeaa8ca91b6b3a38595f12ec45f68b/scripts/codespell.sh#L5-L6

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

**Review by c3y1huang — APPROVED** [review]

**Comment by brandond** [comment]
Can you add this as a CI test, or does it turn up a lot of false positives?

**Comment by jenting** [comment]
> Can you add this as a CI test, or does it turn up a lot of false positives?

Sure, one question is could the k3s project use the GitHub CI action https://github.com/codespell-project/actions-codespell?
Or the k3s project CI should run at the Drone CI only?

File: scripts/codespell.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -e
+
+# Ignore vendor folder and check file names as well
+# Note: ignore "ba" in https://github.com/k3s-io/k3s/blob/4317a91/scripts/provision/vagrant#L54
+codespell --skip=.git,./vendor --check-filenames --ignore-words-list=ba
+
+res=`echo $?`
+if [ "$res" = "0" ]; then
```
Comment (brandond) [inline scripts/codespell.sh:9]: Convention here is:

```suggestion
code=$?
if [ $code -ne 0 ]; then
```

**Review by brandond — CHANGES_REQUESTED** [review]
LGTM, other than one style nit!

**Review by brandond — APPROVED** [review]

**Review by c3y1huang — APPROVED** [review]

File: scripts/codespell.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -e
+
+# Ignore vendor folder and check file names as well
+# Note: ignore "ba" in https://github.com/k3s-io/k3s/blob/4317a91/scripts/provision/vagrant#L54
+codespell --skip=.git,./vendor --check-filenames --ignore-words-list=ba
+
+code=$?
+if [ $code -ne 0 ]; then
+  echo "Error: codespell found one or moe problems!"
```
Comment (bl-ue) [inline scripts/codespell.sh:10]: @jenting is there any way you _meant_ to do this `moe` typo?

**Review by bl-ue — COMMENTED** [review]
@jenting there's a typo the error message alerting users about typos...is this intentional? 😃

File: scripts/codespell.sh (hunk 2 above)
Comment (bl-ue) [reply scripts/codespell.sh:10]: https://github.com/k3s-io/k3s/commit/57041f02392eb7c7c3c72423b98dbb6717d2da55#r45485230

Comment (jenting) [reply scripts/codespell.sh:10]: my bad, good catch!

Comment (bl-ue) [reply scripts/codespell.sh:10]: I know this PR was merged.  It has 3 reviewers, and apparently they all missed this?!?!

**Pull request merged**